napari/layers/image/image.py

```diff
@@ -475,51 +475,43 @@ def _set_view_slice(self):
             )
             indices[not_disp] = downsampled_indices
 
-            disp_shape = self.level_shapes[level, self.dims.displayed]
             scale = np.ones(self.ndim)
             for d in self.dims.displayed:
                 scale[d] = self.level_downsamples[self.data_level][d]
             self._transforms['tile2data'].scale = scale
 
-            if np.any(disp_shape > self._max_tile_shape):
-                for d in self.dims.displayed:
-                    indices[d] = slice(
-                        self._top_left[d],
-                        self._top_left[d] + self._max_tile_shape,
-                        1,
-                    )
-                # Note that top left marks the location of top left canvas
-                # pixel in data coordinates
-                self._transforms['tile2data'].translate = (
-                    self._top_left
-                    * self._transforms['data2world'].scale
-                    * self._transforms['tile2data'].scale
+            for d in self.dims.displayed:
+                indices[d] = slice(
+                    self.corner_pixels[0, d], self.corner_pixels[1, d] + 1, 1,
                 )
-            else:
-                self._transforms['tile2data'].translate = np.zeros(self.ndim)
+            self._transforms['tile2data'].translate = (
+                self.corner_pixels[0]
+                * self._transforms['data2world'].scale
+                * self._transforms['tile2data'].scale
+            )
 
             image = np.asarray(
                 self._data_pyramid[level][tuple(indices)]
             ).transpose(order)
 
-            if level == len(self._data_pyramid) - 1:
-                thumbnail = image
-            else:
-                # Slice thumbnail
-                indices = np.array(self.dims.indices)
-                downsampled_indices = (
-                    indices[not_disp] / self.level_downsamples[-1, not_disp]
-                )
-                downsampled_indices = np.round(
-                    downsampled_indices.astype(float)
-                ).astype(int)
-                downsampled_indices = np.clip(
-                    downsampled_indices, 0, self.level_shapes[-1, not_disp] - 1
-                )
-                indices[not_disp] = downsampled_indices
-                thumbnail = np.asarray(
-                    self._data_pyramid[-1][tuple(indices)]
-                ).transpose(order)
+            if np.product(image.shape) == 0:
+                image = np.zeros((1, 1))
+
+            # Slice thumbnail
+            indices = np.array(self.dims.indices)
+            downsampled_indices = (
+                indices[not_disp] / self.level_downsamples[-1, not_disp]
+            )
+            downsampled_indices = np.round(
+                downsampled_indices.astype(float)
```

Comment:
`.astype(float)` is not needed here: division always results in float in Py3.

Reply:
removed here and above

Reply:
actually this is needed - i got errors when removed

```python-traceback
napari/layers/image/_tests/test_pyramid.py:164:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
napari/layers/image/image.py:228: in __init__
    self._update_dims()
napari/layers/base/base.py:388: in _update_dims
    self.refresh()
napari/layers/base/base.py:605: in refresh
    self._set_view_slice()
napari/layers/image/image.py:470: in _set_view_slice
    downsampled_indices = np.round(downsampled_indices).astype(int)
<__array_function__ internals>:6: in round_
    ???
../../opt/anaconda3/lib/python3.7/site-packages/numpy/core/fromnumeric.py:3599: in round_
    return around(a, decimals=decimals, out=out)
<__array_function__ internals>:6: in around
    ???
../../opt/anaconda3/lib/python3.7/site-packages/numpy/core/fromnumeric.py:3224: in around
    return _wrapfunc(a, 'round', decimals=decimals, out=out)
../../opt/anaconda3/lib/python3.7/site-packages/numpy/core/fromnumeric.py:70: in _wrapfunc
    return _wrapit(obj, method, *args, **kwds)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

obj = array([0.0], dtype=object), method = 'round', args = (), kwds = {'decimals': 0, 'out': None}, wrap = <built-in method __array_wrap__ of numpy.ndarray object at 0x1a373a7940>

    def _wrapit(obj, method, *args, **kwds):
        try:
            wrap = obj.__array_wrap__
        except AttributeError:
            wrap = None
>       result = getattr(asarray(obj), method)(*args, **kwds)
E       TypeError: loop of ufunc does not support argument 0 of type numpy.float64 which has no callable rint method

../../opt/anaconda3/lib/python3.7/site-packages/numpy/core/fromnumeric.py:47: TypeError
```

Reply:
i have added it back in

Reply:
Weird! 🤷‍♂️ sorry for the wild goose chase.